[GPUP][MSE] networkState doesn't move to NETWORK_IDLE once MediaSource.endOfStream() is called

jyavenard opened the pull request:
#### 39a3835b1f4151f2c7cf011e2f5df382b7a7e6b7
<pre>
[GPUP][MSE] networkState doesn&apos;t move to NETWORK_IDLE once MediaSource.endOfStream() is called
<a href="https://bugs.webkit.org/show_bug.cgi?id=254324">https://bugs.webkit.org/show_bug.cgi?id=254324</a>
rdar://107129124

Reviewed by Youenn Fablet.

The required methods weren&apos;t implemented in the
MediaSourcePrivateRemote/RemoteMediaSourceProxy combo of objects.
We implement them.

Covered in media/media-source/media-source-end-of-stream.html once
webkit.org/b/225367 is done.

* LayoutTests/media/media-source/media-source-end-of-stream.html: Fix whitespaces
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::addTrackBuffer): Whitespaces fix
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
(WebKit::RemoteMediaSourceProxy::markEndOfStream):
(WebKit::RemoteMediaSourceProxy::unmarkEndOfStream):
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::isEnded const):
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:

Canonical link: <a href="https://commits.webkit.org/262016@main">https://commits.webkit.org/262016@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c9d21ce24b0eda7931368d2ad1c443f07cfbf35

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/292 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/307 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/319 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/296 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/274 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/292 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/301 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/320 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/297 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/249 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/290 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/321 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/283 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/257 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/299 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/280 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/258 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/250 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/292 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/66 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/284 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
```diff
@@ -125,20 +125,22 @@ void MediaSourcePrivateRemote::bufferedChanged(const PlatformTimeRanges& buffere
 {
-    notImplemented();
-    return false;
+    return m_ended;
```
If all implementations of markEndOfStream/unmarkEndOfStream/isEnded are doing the same with regards to m_ended, maybe this should be moved to MediaSourcePrivate and we would have markEndOfStreamInternal/unmarkEndOfStreamInternal methods that we would specialize

jyavenard replied:
good point. I'll check other implementations and do that if possible.

jyavenard replied:
MediaSourcePrivate is currently a pure virtual class, I don't really want to change this at this stage.

youennf commented on Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
```diff
@@ -125,20 +125,22 @@ void MediaSourcePrivateRemote::bufferedChanged(const PlatformTimeRanges& buffere
 void MediaSourcePrivateRemote::unmarkEndOfStream()
 {
-    notImplemented();
+    // FIXME(125159): implement unmarkEndOfStream()
```
Comment probably not worth it, the FIXME should be in the implementation class?

jyavenard committed:
[GPUP][MSE] networkState doesn't move to NETWORK_IDLE once MediaSource.endOfStream() is called
https://bugs.webkit.org/show_bug.cgi?id=254324
rdar://107129124

Reviewed by Youenn Fablet.

The required methods weren't implemented in the
MediaSourcePrivateRemote/RemoteMediaSourceProxy combo of objects.
We implement them.

Covered in media/media-source/media-source-end-of-stream.html once
webkit.org/b/225367 is done.

* LayoutTests/media/media-source/media-source-end-of-stream.html: Fix whitespaces
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::addTrackBuffer): Whitespaces fix
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
(WebKit::RemoteMediaSourceProxy::markEndOfStream):
(WebKit::RemoteMediaSourceProxy::unmarkEndOfStream):
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::isEnded const):
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:

Canonical link: https://commits.webkit.org/262016@main
